### PR TITLE
Added GET API method for Student and Job Post

### PR DESCRIPTION
**Pull request by emilsangil**
Added the following:
- Student GET API
- Job Posting GET API
- Added some validation for response

Update in the other files
- change the naming of interface from Uppercase first letter to Lowercase
- change the explicit data coming from the frontend into an interface object

### CODE REVIEW

File: src/pages/api/jobPosting/[id]/index.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// Third-party import
+import { NextApiRequest, NextApiResponse } from "next";
+// Local import
+import { getJobPosting } from "@/backend/actions/jobPosting";
+
+export default async function hanlder(req: NextApiRequest, res: NextApiResponse) {
```
Comment (LorenzCastillo) [inline src/pages/api/jobPosting/[id]/index.ts:6]: Handler is misspelled 0_0

File: src/pages/api/jobPosting/getAll.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// Third-party import
+import { NextApiRequest, NextApiResponse } from "next";
+// Local import
+import { getAllPosting } from "@/backend/actions/jobPosting";
+
+export default async function hanlder(req: NextApiRequest, res: NextApiResponse) {
```
Comment (LorenzCastillo) [inline src/pages/api/jobPosting/getAll.ts:6]: Same thing here too but not sure if it's on purpose

Comment (arejandro) [reply src/pages/api/jobPosting/getAll.ts:6]: do it once, its a mistake, do it twice, its intended

File: src/pages/api/jobPosting/getAll.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    if(req.method === "GET") {
+        try {
+
```
Comment (LorenzCastillo) [inline src/pages/api/jobPosting/getAll.ts:9]: Also extra space here if you wanna make it cleaner :)

File: src/pages/api/jobPosting/[id]/index.ts (hunk 1 above)
Comment (emilsangil) [reply src/pages/api/jobPosting/[id]/index.ts]: sorry my bad and nice catch

File: src/pages/api/jobPosting/getAll.ts (hunk 2 above)
Comment (emilsangil) [reply src/pages/api/jobPosting/getAll.ts]: sorry my bad and nice catch

Comment (LorenzCastillo) [reply src/pages/api/jobPosting/getAll.ts]: ur bad at spelling

**Review by LorenzCastillo — APPROVED** [review]
Looks good to me :D

**Review by arejandro — APPROVED** [review]
LGTM

**Pull request merged**